components/image/slider/src/index.js

```diff
@@ -1,45 +1,73 @@
-        : slides
-      }
-    </div>
-  )
-}
+class ImageSlider extends Component {
+  constructor (props) {
+    super(props)
+    const {sliderOptions} = props
+    // In order to accept a custom doAfterSlide callback, and to avoid altering props.sliderOptions
+    // sliderOptions is deeply cloned.
+    this._sliderOptions = cloneDeep(sliderOptions)
+    this._sliderOptions.doAfterSlide = (currentSlide) => {
+      this.setState(currentSlide)
+      sliderOptions.doAfterSlide && sliderOptions.doAfterSlide(currentSlide)
+    }
+  }
+
+  state = {currentSlide: 0}
+
+  render () {
+    const {images, linkFactory, handleClick, dynamicContent, enableCounter} = this.props
+    const slides = this._getSlides(images, linkFactory)
+    return (
+      (slides.length > 0) &&
+      <div onClick={handleClick} className='sui-ImageSlider'>
+        { (slides.length > 1)
+          ? (<ReactSlidy {...this._sliderOptions} dynamicContent={dynamicContent}>{ slides }</ReactSlidy>)
+          : slides
+        }
+        {enableCounter && this._buildCounter(slides.length)}
+      </div>
+    )
+  }
 
-/**
- * Says if there are more than one image in the given list.
- * Implemented for semantic purposes.
- * @param {Array} images List given by props.images.
- * @return {boolean}
- */
-const hasMoreThanOneImage = (images) => (images && images.length > 1)
+  _buildCounter (totalImages) {
+    const classNames = cx('sui-ImageSlider-counter', `sui-ImageSlider-counter--${this.props.counterPosition}`)
+    const Icon = this.props.counterIcon
+    return <div className={classNames}>
+      <Icon svgClass='sui-ImageSlider-counterIcon' />
+      <span className='sui-ImageSlider-counterText'>{this.state.currentSlide + 1}/{totalImages}</span>
+    </div>
+  }
 
-/**
- * @param {Array} images List given by props.images.
- * @return {Array} List of img elements.
- */
-const getSlides = (images, linkFactory) => {
-  if (images && images.length) {
-    return images.map((image, index) => {
-      const key = image.key ? image.key + index : index
-      const img = <img className='sui-ImageSlider-image' key={key} src={image.src} alt={image.alt} />
-      const target = image.target ? image.target : '_blank'
-      return image.link ? linkFactory({ href: image.link, target: target, className: '', children: img, key: key }) : img
-    })
-  } else {
-    return []
+  /**
+   * @param {Array} images List given by props.images.
+   * @return {Array} List of img elements.
+   */
+  _getSlides (images, linkFactory) {
+    if (images && images.length) {
+      return images.map((image, index) => {
+        const key = image.key ? image.key + index : index
+        const img = <img className='sui-ImageSlider-image' key={key} src={image.src} alt={image.alt} />
+        const target = image.target ? image.target : '_blank'
+        return image.link ? linkFactory({ href: image.link, target: target, className: '', children: img, key: key }) : img
```

Comment:
Personal taste, but define objects with more than one property in one line is toooooo ugly, unreadable...

Reply:
I do this kind of things just to annoy people 🤣 
👍 I'll change it